### PR TITLE
Add an Alpine console image

### PR DESCRIPTION
**Pull request by SvenDowideit**
Signed-off-by: Sven Dowideit SvenDowideit@home.org.au

plus a little docs :)

for https://github.com/rancher/os/issues/1151

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,6 +5,34 @@ Docker Images for RancherOS
...
+build can occur.
+
+## Special Entrypoint and CMD
+
+Console images also use special ENTRYPOINT and CMD settings, which are bind mounted
+into the container at run time:
+
+```
+ENTRYPOINT ["/usr/bin/ros", "entrypoint"]
+CMD ["/usr/sbin/console.sh"]
```
Comment (joshwget) [inline README.md]: This was switched to `ros console-init` and a `CMD` isn't specified in the image anymore. It's located in the service file instead (https://github.com/rancher/os-services/blob/master/d/debian.yml#L3). Maybe we can move these docs to a separate PR?

File: images/10-alpineconsole/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
+FROM rancher/os-alpineconsole-base
+# FROM amd64=alpine:3.4 arm64=skip arm=skip
+RUN apk --no-cache add bash openssh iptables rsync sudo vim less curl ca-certificates htop kmod iproute2 util-linux
+# locales psmisc
+RUN rm -rf /etc/ssh/*key*
+RUN rm -fr /sbin/poweroff  /sbin/shutdown /sbin/reboot /sbin/halt /usr/sbin/poweroff  /usr/sbin/shutdown /usr/sbin/reboot /usr/sbin/halt
+#RUN ln -s /sbin/getty /sbin/agetty
+COPY build/update-ssh-keys build/rancheros-install /usr/sbin/
```
Comment (joshwget) [inline images/10-alpineconsole/Dockerfile]: Can you remove `rancheros-install`?

File: images/10-alpineconsole/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    addgroup -g 1101 docker && \
+    addgroup -S sudo && \
+    adduser -D -u 1100 -g rancher -G docker -G sudo -s /bin/bash rancher && \
+    adduser -D -u 1101 -g docker -G docker -G sudo -s /bin/bash docker && \
+    echo ClientAliveInterval 180 >> /etc/ssh/sshd_config && \
+    echo '## allow password less for rancher user' >> /etc/sudoers && \
+    echo 'rancher ALL=(ALL) NOPASSWD: ALL' >> /etc/sudoers && \
+    echo '## allow password less for docker user' >> /etc/sudoers && \
+    echo 'docker ALL=(ALL) NOPASSWD: ALL' >> /etc/sudoers && \
+    ln -sf /usr/bin/docker.dist /usr/bin/docker && \
```
Comment (joshwget) [inline images/10-alpineconsole/Dockerfile]: These symlinks were also removed.

File: images/10-alpineconsole/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    ln -sf /usr/bin/docker-containerd.dist /usr/bin/docker-containerd && \
+    ln -sf /usr/bin/docker-containerd-shim.dist /usr/bin/docker-containerd-shim && \
+    ln -sf /usr/bin/docker-runc.dist /usr/bin/docker-runc \
+    echo 'ssh-keygen -A' >> /usr/sbin/update-ssh-keys
+ENTRYPOINT ["/usr/bin/ros", "entrypoint"]
+CMD ["/usr/sbin/console.sh"]
```
Comment (joshwget) [inline images/10-alpineconsole/Dockerfile]: As mentioned above, no more `CMD` in console images.

**Comment by SvenDowideit** [comment]
@joshwget yup, I've held off updating this, but will do it today.

File: README.md [hunk 5]
```diff
@@ -5,6 +5,37 @@ Docker Images for RancherOS
...
+```
+FROM rancher/os-centosconsole-base
+# FROM amd64=centos:7 arm64=skip arm=skip
+```
+
+The `rancher/os-centosconsole-base` does not actually exist. Dapper will download the
+arch specific image listed in the commented out `# FROM` line, and tag that, so the
+build can occur.
+
+## Special Entrypoint and CMD
```
Comment (joshwget) [inline README.md:26]: I really like the fact that we're documenting this, but this README feels like the wrong place for it. This would go really well in a "custom console image" page in our docs.

File: README.md [hunk 6]
```diff
@@ -5,6 +5,37 @@ Docker Images for RancherOS
...
+
+Console images also use special ENTRYPOINT and CMD settings, which are bind mounted
+into the container at run time:
+
+```
+ENTRYPOINT ["/usr/bin/ros", "entrypoint"]
+```
+
+## Misc details
```
Comment (joshwget) [inline README.md:35]: Same comment as above

File: images/10-alpineconsole/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,23 @@
+FROM rancher/os-alpineconsole-base
+# FROM amd64=alpine:3.4 arm64=skip arm=skip
+RUN apk --no-cache add bash openssh iptables rsync sudo vim less curl ca-certificates htop kmod iproute2 util-linux
```
Comment (joshwget) [inline images/10-alpineconsole/Dockerfile:3]: Do we not need `-U` here? Does `--no-cache` negate the need for it?

Comment (SvenDowideit) [reply images/10-alpineconsole/Dockerfile:3]: yup, --no-cache means it'll always fetch, and then discard.

Comment (joshwget) [reply images/10-alpineconsole/Dockerfile:3]: This is nice. I like the UX of `apk` a lot.

File: README.md (hunk 5 above)
Comment (SvenDowideit) [reply README.md:26]: yup, these docs will be copied there, and we can de-dupe when they go live next release.

Comment (joshwget) [reply README.md:26]: Okay, SGTM.

**Comment by joshwget** [comment]
LGTM. Would be cool if you could take out the commented out lines though.

**Comment by SvenDowideit** [comment]
added info reminding us what they mean instead :)

**Comment by deniseschannon** [comment]
@SvenDowideit Where are the docs update for this new console?

**Comment by SvenDowideit** [comment]
yes, i'm working on them now.
